### ISSUE TITLE
resolver: spurious regression in resolution of globals

### ISSUE THREAD

**Issue by alandonovan** [google/starlark-go#214]
```
$ starlark  # at revision 6ddc71c
Welcome to Starlark (go.starlark.net)
>>> x = 1
>>> x += 2
Traceback (most recent call last):
  <stdin>:1:1: in <toplevel>
Error: global variable x referenced before assignment
```

I suspect this is a REPL bug: the first statement creates x=1 in the globals; the REPL then adds x=1 in the predeclared for the second statement. The x += 1 is resolved as a global because it's an assignment, but there is no global x. Perhaps x +=1 should not be treated as a binding since the binding must already exist.

**Comment by AlekSi** [comment]
It works as expected on 50ca820fafb9

**Comment by alandonovan** [comment]
This was issue #224, fixed by 77c1099ca038b502b6f4607b00edae412647ba37.
